sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp

```diff
@@ -276,6 +301,38 @@ namespace Azure { namespace Core { namespace Test {
           Sanitize(testNameInfo->test_suite_name()), Sanitize(testNameInfo->name()));
       m_testContext.RecordingPath = recordingPath;
       m_interceptor = std::make_unique<Azure::Core::Test::InterceptorManager>(m_testContext);
+
+      if (!m_testContext.IsPlaybackMode())
+      {
+        auto SetBuiltinEnvironment = [](std::string const& targetVariable) {
+          std::string azureVariable = "AZURE" + targetVariable;
+          if (Azure::Core::_internal::Environment::GetVariable(azureVariable.c_str()).empty())
+          {
+            std::string serviceName
+                = Azure::Core::_internal::Environment::GetVariable("AZURE_SERVICE_DIRECTORY");
+            if (serviceName.empty())
+            {
+              throw std::runtime_error(
+                  "Could not find a value for AZURE_" + targetVariable
+                  + " and AZURE_SERVICE_DIRECTORY was not defined. Define either AZURE_"
+                  + targetVariable + " or AZURE_SERVICE_DIRECTORY to resolve.");
+            }
+            // Upper case the serviceName environment variable because all ci.yml environment
+            // variables are upper cased.
+            serviceName = Azure::Core::_internal::StringExtensions::ToUpper(serviceName);
+            std::string targetValue = Azure::Core::_internal::Environment::GetVariable(
+                (serviceName + targetVariable).c_str());
+            if (!targetValue.empty())
+            {
+              Azure::Core::_internal::Environment::SetVariable(
+                  azureVariable.c_str(), targetValue.c_str());
+            }
+          }
+        };
+        SetBuiltinEnvironment("_TENANT_ID");
+        SetBuiltinEnvironment("_CLIENT_ID");
+        SetBuiltinEnvironment("_CLIENT_SECRET");
```

Comment:
There's one little details still making me wonder...  

- We are setting env vars with the tests, which is never something we should :(
- We are setting tenant, client and secret on SetUpBaseTest.  What if the test doesn't even need this values?  It seems risky to me to set this values whenever it is not playback mode (record & live).
- We are setting env vars and we are not unsetting them, I don't think that's good.

Question ->  Instead of adding this logic to `SetUpBaseTest`, can we add it to `GetEnv()` ?  So it is when the test try to get `AZURE_TENANT_ID` when we run this logic and fallback to `ATTESTATION_TENANT_ID` ?  And we don't set the env var, but only return the value of

Reply:
Good points. I thought about adding it to `GetEnv` (actually that was my first implementation - I decided to follow the python pattern because it resulted in a simpler experience), But moving the logic to GetEnv makes sense. It will be slightly more complicated, but reasonable.

FWIW, as far as unsetting environment variables - environment variables are per-process, there is no durable effect outside the current process.